pkg/clusteragent/admission/controllers/webhook/common.go

```diff
@@ -8,13 +8,54 @@
 package webhook
 
 import (
+	"encoding/json"
+	"fmt"
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/common"
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate"
 	"github.com/DataDog/datadog-agent/pkg/config"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// buildAgentSidecarObjectSelectors returns the mutating webhooks object selector based on the configuration
+func buildAgentSidecarObjectSelectors() (namespaceSelector, objectSelector *metav1.LabelSelector) {
+
+	// Read and parse label selectors
+	labelSelectorsJSON := config.Datadog.GetString("admission_controller.agent_sidecar.label_selectors")
+
+	var labelSelectors []map[string]string
+
+	err := json.Unmarshal([]byte(labelSelectorsJSON), &labelSelectors)
+	if err != nil {
+		fmt.Printf("failed to parse label selectors for admission controller agent sidecar injection webhook: %s", err)
+		return nil, nil
+	}
+
+	// Read and parse namespace selectors
+	nsSelectorsJSON := config.Datadog.GetString("admission_controller.agent_sidecar.namespace_selectors")
+	var nsSelectors []map[string]string
+
+	err = json.Unmarshal([]byte(nsSelectorsJSON), &nsSelectors)
+	if err != nil {
+		fmt.Printf("failed to parse namespace selectors for admission controller agent sidecar injection webhook: %s", err)
+		return nil, nil
+	}
+
+	if len(labelSelectors) > 0 {
+		objectSelector = &metav1.LabelSelector{
+			MatchLabels: labelSelectors[0],
```

Comment:
Why only the first one?

Reply:
In the first version of this feature, we allow configuring at most 1 custom selector.

In a future version, we will allow setting several selectors and configuring overrides for each of them. 

We are defining the selector config as a slice of selectors in order not to break user setups in the future when we upgrade to version 2, while constraining selectors to only 1 selector in version 1.